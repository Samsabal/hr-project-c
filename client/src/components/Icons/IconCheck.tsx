import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconCheck = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M20 6L9 17L4 12" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
