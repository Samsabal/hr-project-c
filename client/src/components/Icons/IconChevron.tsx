import React from 'react'

import { IconProps } from '../../types/IconProps'

export const IconChevron = ({ size, fill, color, strokeWidth = 2, direction }: IconProps) => {
    switch (direction) {
        case 'down':
            return (
                <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
                    <path d="M6 9L12 15L18 9" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
                </svg>
            );
        case 'up':
            return (
                <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
                    <path d="M18 15L12 9L6 15" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
                </svg>
            );
    }
    // Return a chevron pointing right by default
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M9 18L15 12L9 6" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    );
}