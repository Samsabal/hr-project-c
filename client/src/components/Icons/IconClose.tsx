import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconClose = ({ size, fill, color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path d="M18 6L6 18M6 6L18 18" strokeWidth={strokeWidth} strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
