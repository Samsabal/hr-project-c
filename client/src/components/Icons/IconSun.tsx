import React from 'react'
import { IconProps } from '../../types/IconProps'

export const IconSun = ({ size, fill = 'none', color, strokeWidth = 2 }: IconProps) => {
    return (
        <svg className={color} width={size} height={size} viewBox="0 0 24 24" fill="none">
            <path opacity="0.12" d="M12 17C14.7614 17 17 14.7614 17 12C17 9.23858 14.7614 7 12 7C9.23858 7 7 9.23858 7 12C7 14.7614 9.23858 17 12 17Z" className={fill} />
            <path d="M12 2V4M12 20V22M4 12H2M6.31412 6.31412L4.8999 4.8999M17.6859 6.31412L19.1001 4.8999M6.31412 17.69L4.8999 19.1042M17.6859 17.69L19.1001 19.1042M22 12H20M17 12C17 14.7614 14.7614 17 12 17C9.23858 17 7 14.7614 7 12C7 9.23858 9.23858 7 12 7C14.7614 7 17 9.23858 17 12Z" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" />
        </svg>
    )
}
