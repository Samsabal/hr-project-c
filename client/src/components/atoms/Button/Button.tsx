import React from 'react'
import { Link } from 'react-router-dom';

type ButtonProps = {
    size: 'small' | 'medium' | 'large',
    width: 'full' | 'content',
    type: 'primary' | 'secondary-gray' | 'secondary-color' | 'tertiary-gray' | 'tertiary-color' | 'error',
    text: string,
    icon?: JSX.Element,
    url?: string,
    onclick?: () => any,
    disabled?: boolean,
    formType?: "submit" | "button" | "reset"
}

export function Button({ size, width, type, text, icon, url, onclick, formType, disabled }: ButtonProps) {
    const buttonWidthStyle = width === 'full' ? "w-full" : "";
    let buttonSizeStyle = "";
    let buttonTypeStyle = "";

    switch (size) {
        case 'large':
            buttonSizeStyle = "py-2.5 px-5 font-semibold";
            break;
        case 'medium':
            buttonSizeStyle = "py-2.5 px-4 font-semibold text-sm";
            break;
        case 'small':
            buttonSizeStyle = "py-2 px-3.5 font-semibold text-sm";
            break;
    }

    switch (type) {
        case 'primary':
            buttonTypeStyle = "shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700";
            break;
        case 'secondary-gray':
            buttonTypeStyle = "shadow-sm text-gray-800 border border-gray-200 dark:border-dark-600 dark:text-white hover:bg-gray-50 dark:hover:bg-dark-700 focus:outline outline-gray-100 dark:outline-dark-600";
            break;
        case 'secondary-color':
            buttonTypeStyle = "text-primary-600 dark:text-white border border-primary-50 dark:border-dark-600 bg-primary-50 dark:bg-dark-600 hover:bg-primary-100";
            break;
        case 'tertiary-gray':
            buttonTypeStyle = "text-gray-800 border border-primary-600 border-opacity-0 hover:bg-gray-50 dark:text-white dark:hover:bg-dark-700";
            break;
        case 'tertiary-color':
            buttonTypeStyle = "text-primary-600 dark:text-gray-300 border border-primary-600 border-opacity-0 hover:bg-primary-50 dark:hover:bg-dark-600";
            break;
        case 'error':
            buttonTypeStyle = "shadow-sm bg-error-600 border border-error-600 text-white hover:bg-error-700 focus:outline outline-error-200 dark:outline-error-500 hover:border-error-700"
    };

    return (
        <>
            {url
                ?
                <Link to={url} className={buttonWidthStyle}>
                    <button disabled={disabled} type={formType} onClick={onclick} className={`${buttonTypeStyle} ${buttonWidthStyle} ${buttonSizeStyle}
                flex justify-center items-center gap-x-2.5 rounded-lg
                ${disabled ? "cursor-not-allowed" : "cursor-pointer"}
                `}>
                        {icon}
                        {text}
                    </button>
                </Link>
                :
                <button disabled={disabled} type={formType} onClick={onclick} className={`${buttonTypeStyle} ${buttonWidthStyle} ${buttonSizeStyle}
                flex justify-center items-center gap-x-2.5 rounded-lg
                ${disabled ? "cursor-not-allowed" : "cursor-pointer"}
                `}>
                    {icon}
                    {text}
                </button>
            }
        </>
    )
}
