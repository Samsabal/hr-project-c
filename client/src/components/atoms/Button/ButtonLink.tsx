import React from 'react'
import { Link } from 'react-router-dom';

export enum ButtonLinkSize {
    MEDIUM,
    LARGE,
}

export enum ButtonLinkType {
    COLOR,
    GRAY
}

type ButtonLinkProps = {
    size: "medium" | "large",
    type: "color" | "gray",
    text: string,
    icon?: JSX.Element,
    url?: string,
    onclick?: () => any
}

export function ButtonLink({ size, type, text, icon, url, onclick }: ButtonLinkProps) {
    const buttonTypeStyle = type === "color" ? "text-primary-600 hover:text-primary-500" : "text-gray-600 hover:text-gray-400 dark:text-dark-300 dark:hover:text-gray-200";
    const buttonSizeStyle = size === "medium" ? "text-sm" : "";

    return (
        <>
            {url
                ?
                <Link to={url}>
                    <button onClick={onclick} className={`${buttonTypeStyle} ${buttonSizeStyle}
                        flex justify-center items-center gap-x-2.5 font-semibold`}>
                        {icon}
                        {text}
                    </button>
                </Link>
                :
                <button onClick={onclick} className={`${buttonTypeStyle} ${buttonSizeStyle}
                flex justify-center items-center gap-x-2 font-semibold`}>
                    {icon}
                    {text}
                </button>
            }
        </>
    )
}
