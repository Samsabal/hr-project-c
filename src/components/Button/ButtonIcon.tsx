import React from 'react'

type ButtonOnlyIconProps = {
    icon: JSX.Element
    onclick: () => any
}

export const ButtonIcon = ({ icon, onclick }: ButtonOnlyIconProps) => {
    return (
        <button onClick={onclick} className='flex h-min items-center justify-center p-2 dark:hover:bg-dark-600 hover:bg-gray-100 rounded-md'>
            {icon}
        </button>
    )
}
