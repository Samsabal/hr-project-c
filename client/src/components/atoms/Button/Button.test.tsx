import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import { Button } from "./Button";
import { IconPlus } from "../Icons/Icons";

describe ("Button", () => {
    test('renders button component correctly and has only required props', () => {
        const container1 = render(
            <Button
            size="small"
            width="full"
            type="primary"
            text="Click me"
            />
        );
        expect(container1).toMatchSnapshot();
        
        const container2 = render(
            <Button
            size="medium"
            width="content"
            type="secondary-gray"
            text="Click me"
            />
        );
        expect(container2).toMatchSnapshot();
        
        const container3 = render(
            <Button
            size="large"
            width="full"
            type="tertiary-color"
            text="Click me"
            />
        );
        expect(container3).toMatchSnapshot();

        const container4 = render(
            <Button
            size="small"
            width="full"
            type="secondary-color"
            text="Click me"
            />
        );
        expect(container4).toMatchSnapshot();

        const container5 = render(
            <Button
            size="small"
            width="full"
            type="tertiary-gray"
            text="Click me"
            />
        );
        expect(container5).toMatchSnapshot();

        const container6 = render(
            <Button
            size="small"
            width="full"
            type="error"
            text="Click me"
            />
        );
        expect(container6).toMatchSnapshot();
    });

    test('renders button component correctly, has required props and tests optional props', () => {
        const mockOnClick = jest.fn();
    
        const { container } = render(
            <Button
            size="small"
            width="full"
            type="primary"
            text="Click me"
            icon={<IconPlus size='20' color='stroke-gray-700 dark:stroke-white' fill='' />}
            // url="/button"
            onclick={mockOnClick}
            disabled={true}
            formType="button"
            />
        );
        expect(container).toMatchSnapshot();

        expect(container.querySelector("Button")).toBeDisabled();
        fireEvent.click(container);
        expect(mockOnClick).not.toHaveBeenCalled();
    });
});