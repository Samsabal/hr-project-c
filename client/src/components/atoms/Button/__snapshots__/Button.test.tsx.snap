// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button component correctly and has only required props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders button component correctly and has only required props 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="bg-white dark:bg-dark-800 shadow-sm text-gray-800 border border-gray-200 dark:border-dark-600 dark:text-white hover:bg-gray-50 dark:hover:bg-dark-700 focus:outline outline-gray-100 dark:outline-dark-600  py-2.5 px-4 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="bg-white dark:bg-dark-800 shadow-sm text-gray-800 border border-gray-200 dark:border-dark-600 dark:text-white hover:bg-gray-50 dark:hover:bg-dark-700 focus:outline outline-gray-100 dark:outline-dark-600  py-2.5 px-4 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders button component correctly and has only required props 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="bg-white dark:bg-dark-800 shadow-sm text-gray-800 border border-gray-200 dark:border-dark-600 dark:text-white hover:bg-gray-50 dark:hover:bg-dark-700 focus:outline outline-gray-100 dark:outline-dark-600  py-2.5 px-4 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-primary-600 dark:text-gray-300 border border-primary-600 border-opacity-0 hover:bg-primary-50 dark:hover:bg-dark-600 w-full py-2.5 px-5 font-semibold
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text-primary-600 dark:text-gray-300 border border-primary-600 border-opacity-0 hover:bg-primary-50 dark:hover:bg-dark-600 w-full py-2.5 px-5 font-semibold
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders button component correctly and has only required props 4`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="bg-white dark:bg-dark-800 shadow-sm text-gray-800 border border-gray-200 dark:border-dark-600 dark:text-white hover:bg-gray-50 dark:hover:bg-dark-700 focus:outline outline-gray-100 dark:outline-dark-600  py-2.5 px-4 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-primary-600 dark:text-gray-300 border border-primary-600 border-opacity-0 hover:bg-primary-50 dark:hover:bg-dark-600 w-full py-2.5 px-5 font-semibold
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-primary-600 dark:text-white border border-primary-50 dark:border-dark-600 bg-primary-50 dark:bg-dark-600 hover:bg-primary-100 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text-primary-600 dark:text-white border border-primary-50 dark:border-dark-600 bg-primary-50 dark:bg-dark-600 hover:bg-primary-100 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders button component correctly and has only required props 5`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="bg-white dark:bg-dark-800 shadow-sm text-gray-800 border border-gray-200 dark:border-dark-600 dark:text-white hover:bg-gray-50 dark:hover:bg-dark-700 focus:outline outline-gray-100 dark:outline-dark-600  py-2.5 px-4 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-primary-600 dark:text-gray-300 border border-primary-600 border-opacity-0 hover:bg-primary-50 dark:hover:bg-dark-600 w-full py-2.5 px-5 font-semibold
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-primary-600 dark:text-white border border-primary-50 dark:border-dark-600 bg-primary-50 dark:bg-dark-600 hover:bg-primary-100 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-gray-800 border border-primary-600 border-opacity-0 hover:bg-gray-100 dark:text-white dark:hover:bg-dark-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="text-gray-800 border border-primary-600 border-opacity-0 hover:bg-gray-100 dark:text-white dark:hover:bg-dark-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders button component correctly and has only required props 6`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="bg-white dark:bg-dark-800 shadow-sm text-gray-800 border border-gray-200 dark:border-dark-600 dark:text-white hover:bg-gray-50 dark:hover:bg-dark-700 focus:outline outline-gray-100 dark:outline-dark-600  py-2.5 px-4 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-primary-600 dark:text-gray-300 border border-primary-600 border-opacity-0 hover:bg-primary-50 dark:hover:bg-dark-600 w-full py-2.5 px-5 font-semibold
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-primary-600 dark:text-white border border-primary-50 dark:border-dark-600 bg-primary-50 dark:bg-dark-600 hover:bg-primary-100 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="text-gray-800 border border-primary-600 border-opacity-0 hover:bg-gray-100 dark:text-white dark:hover:bg-dark-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
    <div>
      <button
        class="shadow-sm bg-error-600 border border-error-600 text-white hover:bg-error-700 focus:outline outline-error-200 dark:outline-error-500 hover:border-error-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
      >
        Click me
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="shadow-sm bg-error-600 border border-error-600 text-white hover:bg-error-700 focus:outline outline-error-200 dark:outline-error-500 hover:border-error-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-pointer
                "
    >
      Click me
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button renders button component correctly, has required props and tests optional props 1`] = `
<div>
  <button
    class="shadow-sm bg-primary-600 border border-primary-600 text-white hover:bg-primary-700 focus:outline outline-primary-200 dark:outline-primary-500 hover:border-primary-700 w-full py-2 px-3.5 font-semibold text-sm
                flex justify-center items-center gap-x-2.5 rounded-lg whitespace-nowrap
                cursor-not-allowed
                "
    disabled=""
    type="button"
  >
    <svg
      class="stroke-gray-700 dark:stroke-white"
      fill="none"
      height="20"
      viewBox="0 0 24 24"
      width="20"
    >
      <path
        d="M12 5V19M5 12H19"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
    Click me
  </button>
</div>
`;
